pydm/main_window.py

```diff
@@ -498,3 +501,59 @@ def _confirm_quit(self, callback):
     def show_macro_window(self):
         macro_window = MacroWindow(self)
         macro_window.show()
+
+    def add_menu_items(self):
+        # create the custom menu with user given items 
+        items = self.display_widget().menu_items()
+        if len(items) == 0:
+            self.ui.menuCustomActions.setEnabled(False)
+        else:
+            self.ui.menuCustomActions.setEnabled(True)
+            self.create_menu(self.ui.menuCustomActions, items)
+        
+        # connect custom save, save as, and load functions
+        file_menu_items = self.display_widget().file_menu_items()
+        if len(file_menu_items) != 0:
+            valid_keys = ('save', 'save_as', 'load')
+            ui_actions = (self.ui.actionSave, self.ui.actionSave_As, self.ui.actionLoad)
+            action_dict = dict(zip(valid_keys, ui_actions))
+            # iterate through user given keys, which need to match the possible keys
+            for key in file_menu_items.keys():
+                fileAction = file_menu_items[key]
+                if key.lower() in valid_keys:
+                    ui_action = action_dict[key.lower()]
+                    ui_action.setEnabled(True)
+                    fileAction = file_menu_items[key]
+                    if isinstance(fileAction, tuple):
+                        shortcut = fileAction[1]
+                        fileAction = fileAction[0]
+                        PyDMMainWindow().add_shortcut(ui_action, shortcut)
+                        ui_action.triggered.connect(fileAction)
```

Comment:
Need to implement the case where fileAction is just a callable rather than a tuple

Reply:
Done